lib/options.go

```diff
@@ -187,6 +189,103 @@ func ParseCIDR(s string) (*IPNet, error) {
 	return &parsedIPNet, nil
 }
 
+// HostnameTrie is a tree-structured list of hostname matches with support
+// for wildcards exclusively at the start of the pattern. Items may only
+// be inserted and searched. Internationalized hostnames are valid.
+type HostnameTrie struct {
```

Comment:
now that you don't have other properties in this struct, you might be able to have something like `type HostnameTrie map[rune]HostnameTrie`, which might resolve the envconfig issue I posted below. Though I'm not sure if this wouldn't cause other issues :man_shrugging:

Reply:
I tried doing something similar when I was working on 372f140 and ran into some issues. Will try again in the next commit :+1: